pipeline-runner/R/gem2s-2-load_user_files.R

```diff
@@ -71,19 +72,38 @@ read_10x_files <- function(config, input_dir) {
       paste(sample_fpaths, collapse = " - ")
     )
 
-    counts <- Seurat::Read10X(sample_dir, gene.column = 1)
+    annot <- read.delim(annot_fpath, header = FALSE)
 
-    if (is(counts, "list")) {
-      slot <- "Gene Expression"
-      # questionable: grab first slot if no gene expression
-      if (!slot %in% names(counts)) slot <- names(counts)[1]
-      counts <- counts[[slot]]
+    if (ncol(annot) == 1 || annot[1, 2] == "Gene Expression") {
+      annot[, 2] <- annot[, 1]
     }
 
-    annot <- read.delim(annot_fpath, header = FALSE)
+    features_types <- extract_features_types(annot)
+
+    message("Features types is ", features_types, "for sample ", sample)
+
+    if (features_types == -1) {
```

Comment:
if `-1` has a meaning like `not_found` considering naming it as a constant, or adding a comment

Reply:
-1 is actually

SYM_IDS

So when we replace with constants it will be ok